tests/jballerina-unit-test/src/test/resources/test-src/query/inner-queries.bal

```diff
@@ -302,6 +302,57 @@ function testTypeTestInWhereClause() {
     assertEquality(120, result[5]);
 }
 
+type BddPath record {|
+    int[] pos = [1];
+|};
+
+public type MappingAlternative record {|
+    int[] pos;
+|};
+
+function testQueryExpWithinSelectClause1() {
+    BddPath[] paths = [{}, {}, {}];
+
+    MappingAlternative[] alts = from var {pos} in paths
+        select {
+            pos: (from var atom in pos
+                select atom)
+        };
+
+    MappingAlternative[] expected = [{"pos": [1]}, {"pos": [1]}, {"pos": [1]}];
+    assertEquality(expected, alts);
+}
+
+function testQueryExpWithinSelectClause2() {
+    int[][] data = [[1, 2, 3, 4, 5]];
+
+    (function () returns int[])[] res = from int[] arr in data
+         select function() returns int[] {
+        int[] evenNumbers = from int i in arr
+                 where i % 2 == 0
+                 select i;
+        return evenNumbers;
+    };
+    function () returns int[] func = res[0];
+    int[] expected = [2, 4];
+    assertEquality(expected, func());
+}
+
+function testQueryExpWithinQueryAction() returns error? {
+    int[][] data = [[2, 3, 4]];
+    check from int[] arr in data
+        do {
+            function () returns int[] func = function() returns int[] {
+                int[] evenNumbers = from int i in arr
+                    where i % 2 == 0
+                    select i;
+                return evenNumbers;
+            };
+            int[] expected = [2, 4];
+            assertEquality(expected, func());
+        };
+}
+
```

Comment:
Shouldn't the following also work?
```ballerina
type A record {|
    int[] pos;
|};

public function main() {
    A[] res = from var a in 2...5
        select {
            pos: (from var b in (from var c in 1...2 where c == a select c) // undefined symbol 'a'
                select b)
        };
}
```

Reply:
Here the issue is of two parts.
1. Parent env is set incorrectly (fixed with 8c79123497c8c7eed5bcf4c9b8536db9966b23b1)
2. Closure issue in nested queries (https://github.com/ballerina-platform/ballerina-lang/issues/32710) which requires QueryDesugar rewrite

Reply:
Shall we put this example in #32710 to track it and address it in that issue?

Reply:
+1